dotnet/src/SemanticKernel.Core/Planning/Plan.cs

```diff
@@ -400,15 +400,13 @@ internal string ExpandFromVariables(ContextVariables variables, string input)
 
             // Execute the step
             var result = await kernel.RunAsync(step, functionVariables, cancellationToken).ConfigureAwait(false);
-
-            var stepResult = result.FunctionResults.FirstOrDefault();
-            if (stepResult is null)
+            if (result is null)
             {
                 // Step was cancelled
                 return null;
```

Comment:
If a function was cancelled or skipped it render's no result.

@matthewbolanos @SergeyMenshykh If we stop using Skip we actually get the benefit of not having the current problem where a plan step don't know when to skip.

Reply:
Yep, will be done separately in the PR related to removing the Skip action.